Editorial Automation: Why & How to Set Up Chat-Ops for your Own Review System on GitHub (#341)
ropensci/roweb3: content/blog/2022-06-02-chatops-review-system-github/index.md

---

At rOpenSci Software Peer-Review, processes have been enhanced with :sparkles: automation :sparkles:. 
For instance, for checking compliance of a software package with some criteria, instead of cloning a git repository locally, installing its dependencies, running the checks, and manually posting the results… we can simply type the following **command** into a GitHub issue comment:

```
@ropensci-review-bot check package
```

Similarly, we can use the following command to register a reviewer in the submission issue metadata (filling YAML data) as well as in our Airtable database.

```
@ropensci-review-bot add @maelle to reviewers
```

> Inspiring?

Reviewer: ```suggestion
Inspired?
```

In this post we shall explain how to assess whether adopting the editorial bot generator buffy, like we did, is the right fit for your system.
We will detail steps involved in setting buffy up for your own needs.

## What is an editorial bot generator?

The editorial bot generator buffy is the tool that supports the aforementioned commands.

> It is a **Ruby codebase** that one can 

Reviewer: ```suggestion
It is a **Ruby codebase** that can 
```
along with suggestions in the following lines ... :smile:

> * **configure** and **extend**, 

Reviewer: ```suggestion
be * **configured** and **extended**, 
```

> * then **deploy** on for instance Heroku where it is on all the time, 

Reviewer: ```suggestion
* then be **deployed** on for instance Heroku where it is on all the time, 
```

* and hook into **GitHub issues or pull requests** via, well, **webhooks**. 🪝

Every time a new issue is opened, every time an issue comment is created, their contents are sent to the deployed app.
If it corresponds to the **regular expression** of a **registered command**, something happens: a check is launched somewhere, a database is updated, information is copied back to the GitHub issue, etc.

> The "visible" bot is the **GitHub account used as bot face**: commands are addressed to it, and a "Personal" Access Token of its is used to post GitHub issue comments.

Reviewer: ```suggestion
The "visible" bot is the [**GitHub account used as bot face**](https://github.com/ropensci-review-bot): commands are addressed to it, and a "Personal" Access Token of its is used to post GitHub issue comments.
```

The initial set of possible commands and actions is the [list](https://buffy.readthedocs.io/en/latest/available_responders.html) of buffy **responders**.
One can add responders by writing buffy code following [documented instructions](https://buffy.readthedocs.io/en/latest/custom_responder.html).
Note that one of the responders allows to launch a GitHub Action Workflow which can circumvent absence of Ruby knowledge on a team -- although a pure Ruby responder would be faster.
You can also send a call to any API, so if you can build an external API, you can really implement many things.

> ## The editorial bot generator buffy is for you if...

Reviewer: ```suggestion
## The editorial bot generator is for you if...
```

---

At rOpenSci Software Peer-Review, processes have been enhanced with :sparkles: automation :sparkles:. 
For instance, for checking compliance of a software package with some criteria, instead of cloning a git repository locally, installing its dependencies, running the checks, and manually posting the results… we can simply type the following **command** into a GitHub issue comment:

```
@ropensci-review-bot check package
```

Similarly, we can use the following command to register a reviewer in the submission issue metadata (filling YAML data) as well as in our Airtable database.

```
@ropensci-review-bot add @maelle to reviewers
```

Inspired?
In this post we shall explain how to assess whether adopting the editorial bot generator buffy, like we did, is the right fit for your system.
We will detail steps involved in setting buffy up for your own needs.

## What is an editorial bot generator?

> The editorial bot generator buffy is the tool that supports the aforementioned commands.

Reviewer 2: ```suggestion
The editorial bot generator [Buffy](https://github.com/openjournals/buffy) is the tool that supports the aforementioned commands.
```

As here it's been introduced I would upcase Buffy and link to its url
